add: firefox support

User-425 opened the pull request:
add firefox support and seperate between the installation and the usage instructions.

User-425 committed:
add: firefox support

qcrao commented:
Great job!

Hi @User-425 ,

Thanks for your PR! I tested it, and it works perfectly in Firefox. However, I encountered an error in Chrome, as shown below:

<img width="1034" alt="image" src="https://github.com/user-attachments/assets/11e37193-7b9f-47b9-bba4-c5d19955dbd0" />

Would it be possible for you to add support for different build commands for different browsers? Specifically, it would be great if we could output different manifest.json files for each browser.

Looking forward to your thoughts on this!

User-425 commented:
@qcrao, you might want to look into this website. 
https://stackoverflow.com/questions/78491335/how-can-i-write-a-chrome-and-firefox-manifest-v3-extension-that-shares-code-when 
look at the comment,  
`Usually it's solved by using a bundler like webpack, but in new Chrome you can specify both scripts and service_worker, the former will be ignored with a warning in unpacked mode, so if you don't need to support the old Chrome you're fine`.
In short, we can ignore it. I just tested it on Chrome, it's functional for me. But I may create a different manifest.json if needed.

qcrao commented:
Hi @User-425,

Got it! In that case, we probably don’t need to add a separate build script. I tested it, and it only shows a warning, so that should be fine.

I’d like to authorize you to publish the extension on the Firefox Add-ons store—would you be willing to handle that?

For now, I won’t merge this MR. Once the updated version is published on both Firefox and Chrome without any issues, I’ll merge the code changes.

Let me know what you think!

Best,
@qcrao

User-425 commented:
Hi again @qcrao  

Sounds good to me. Let me handle that now. I'll let you know when it's done. Thanks.

User-425 committed:
update firefox installation to include firefox extension page and removed some "chrome" keywords

qcrao commented:
> Hello @qcrao, the extension finally listed on the store. You might want to check it out. https://addons.mozilla.org/en-US/firefox/addon/chatgpt-bulk-delete/ Let me know if you need anything! Thank you.

Hi @User-425,

Thanks for your efforts in adding Firefox support! I just tested the latest version, but unfortunately, it doesn’t seem to work correctly. From what I can see, the code is currently at version 5.8, while it should be 5.9 to function properly.

Could you check this and update accordingly? Let me know if you need any help!

Best,
@qcrao

User-425 committed:
Merge https://github.com/qcrao/bulk-delete-chatGPT

User-425 committed:
update icon48.png into a square version for firefox requirement

User-425 commented:
Hi @qcrao, I just updated the extension. Firefox need like 1x24 hour to do daily update, so it may take some time to roll out for all user. Or, the user can update it manually through the settings. 

And, I just updated the icon48.png into a square version. The ratio of the image still the same, I just added some whitespace in order to meet Firefox requirement. Please check whether it is compatible with Chrome or not. If not, let me just PR it into Firefox branch, and I'll maintain the extension in that branch.

qcrao commented:
<img width="114" alt="image" src="https://github.com/user-attachments/assets/1723c647-c0bf-4145-be8d-d279a45218bf" />

👆master | this pr

Hi @User-425, I just checked version 5.9 on Firefox, and it’s working well now! Thanks for the update.

However, the icon looks a bit off on Chrome. Keeping a separate branch for the Firefox version sounds like a good plan.

I’ve added you as a collaborator on the GitHub project, please check your email. Let me know if you need anything!
